### PR TITLE
BUG: Sort index for series in selection for context adjustment for pandas backend

### PR DESCRIPTION
**Pull request by LeeTZ**
Overview
=======
With time context adjustment enabled, the resulting series for window could be trimmed by time context.  In window execution, data is expanded to a larger context, to include more rows. And the trimmed result, which is a Series, may have an altered index. 

For a `table.execute()`:
```
time                  id        value
2021-01-01       1          1.0
2021-01-02       2          2.0
2021-01-03       3          3.0
2021-01-04       4          4.0
```
When we execute this with a trimmed context, and trailing window for 1 day look back:
```
win = ibis.window(preceding=1, following=0, order_by=table.id)
expr =table.mutate(mean_col=table.value.mean().over(win))
expr.execute(pd.Timestamp('2021-01-03'), pd.Timestamp('2021-01-04'))
```

In `execute` in `Selection`, we are concatenating Dataframe and the resulting Series of window, To make sure rows in Series and the Dataframe are aligned properly, we first `sort_index` each data pieces and then assign data index manually to series.

For this example, in `execute_selection_dataframe`: 
```
>>> data
   time                  id        value
0 2021-01-03       3          3.0
1 2021-01-04       4          4.0
[2 rows x 3 columns]

>> mean_col
2  2.5
3  3.5
Name: mean_col, Length: 2, dtype:float64
```

And when we concat the resulting Series to dataframe, there will be `NaT` and `NaN` inserted since indexes don't match:
```
   time                  id        value    mean_col
0 2021-01-03       3          3.0         NaN
1 2021-01-04       4          4.0         NaN
2 NaT                  NaN     NaN        2.5
3 NaT                  NaN     NaN        3.5
```
Test
===
New test case is added in `ibis/backends/tests/test_timecontext.py` for a grouby window with time context

### CODE REVIEW

**Comment by icexelloss** [comment]
Can you show the index change before and after this?

**Comment by icexelloss** [comment]
LGTM. Please include an example in the overview explaining the index before and after

File: ibis/backends/tests/test_timecontext.py [hunk 1]
```diff
@@ -75,3 +75,31 @@ def test_context_adjustment_filter_before_window(alltypes, df, context):
         expected = expected.reset_index(drop=True)
 
         tm.assert_frame_equal(result, expected)
+
+
+@pytest.mark.only_on_backends(['pandas', 'pyspark'])
+def test_grouped_bounded_expanding_window(
```
Comment (icexelloss) [inline ibis/backends/tests/test_timecontext.py:81]: How does this test cover your change? Isn't the context not adjusted for expanding window?

Comment (icexelloss) [reply ibis/backends/tests/test_timecontext.py:81]: Discussed offline. 

@LeeTZ Can you verify that existing context adjustment tests fail without the change in selection.py?

**Comment by jreback** [comment]
looks fine, @icexelloss when ready

**Review by jreback — APPROVED** [review]

**Comment by jreback** [comment]
thanks @LeeTZ

**Pull request merged**